src/openrct2/management/research.c

```diff
@@ -208,12 +225,44 @@ void research_finish_item(sint32 entryIndex)
                 }
             }
 
-            set_format_arg(0, rct_string_id, ((rideEntry->flags & RIDE_ENTRY_FLAG_SEPARATE_RIDE_NAME)) ?
-                rideEntry->name : RideNaming[base_ride_type].name);
+            if (!gConfigInterface.select_by_track_type)
+            {
+                availabilityString = STR_NEWS_ITEM_RESEARCH_NEW_RIDE_AVAILABLE;
+
+                set_format_arg(0, rct_string_id, ((rideEntry->flags & RIDE_ENTRY_FLAG_SEPARATE_RIDE_NAME)) ?
+                               rideEntry->name : get_friendly_track_type_name(base_ride_type, rideEntry));    // Makes sure the correct track name is displayed,
```

Comment:
Can you move this inline comment above the call? I understand it concerns the condition, but also makes this line overly long. If needed, you can extract the condition result to another variable.